src/helpers/files.js

```diff
@@ -171,22 +206,24 @@ const FilesWorkspacePlugin = {
 			const vm = new View({
 				propsData: {
 					path: fileList.getCurrentDirectory(),
+					hasDescriptionFile: !!fileList.findFile(descriptionFile)
 				},
 				store,
 			}).$mount(this.el)
-
 			subscribe('files:navigation:changed', () => {
 				// Expose if the default file list is active to the component
 				// to only render the workspace if the file list is actually visible
 				vm.active = OCA.Files.App.getCurrentFileList() === fileList
 			})
-
 			fileList.$el.on('urlChanged', data => {
 				vm.path = data.dir.toString()
 			})
 			fileList.$el.on('changeDirectory', data => {
 				vm.path = data.dir.toString()
 			})
+			fileList.$table.on('DOMSubtreeModified', () => {
```

Comment:
```suggestion
			fileList.$table.on('afterChangeDirectory', () => {
```

I think this should also work and is only triggered when the list of files is actually updated

Reply:
This event is not triggered when we add the README.md file to the home directory. Do you have any alternative way?

Reply:
That should be fine if we rely on the existing mechanism to check for a readme existence in the component as noted below https://github.com/nextcloud/text/pull/3291#discussion_r1043361054